jmh-core/src/main/java/org/openjdk/jmh/generators/core/MethodGroup.java

```diff
@@ -72,20 +72,17 @@ public int compareTo(MethodGroup o) {
 
     public void addMethod(MethodInfo method, int threads) {
         MethodInvocation mi = new MethodInvocation(method, threads);
-        MethodInvocation exist = methods.get(mi);
-        if (exist == null) {
-            methods.put(mi, mi);
-        } else {
+        if (!methods.add(mi)) {
             throw new GenerationException(
                     "@" + Benchmark.class.getSimpleName() + " method is duplicate with " +
-                           exist.method.getQualifiedName() + ". JMH needs an uniquely named method, regardless of the arguments list. ",
+                            mi.method.getQualifiedName() + ". JMH needs an uniquely named method, regardless of the arguments list. ",
```

Comment:
Aha! Now I remember why it was `Map` -- we needed `exist` here. But we can also simplify the diagnostics. Say, for example:

```
            throw new GenerationException(
                    "Duplicate @" + Benchmark.class.getSimpleName() + " method name: " +
                            mi.method.getQualifiedName() + ". JMH needs an uniquely named method, regardless of the arguments list. ",
```

This would probably need test adjustments that might expect different error message.

Reply:
I found a corresponding test case `jmh-core-ct/src/test/java/org/openjdk/jmh/ct/benchmark/args/ArgumentListAmbiguityTest`. Any other test cases do you think need to be adjusted?

Reply:
Current build and GHA run full suite of tests. So, if no other tests are failing, no other tests need adjustment.

Reply:
I am running the tests locally to find the failed tests. I will push the new code after all the tests passed.

Reply:
Code updated. And no tests need to be adjusted.